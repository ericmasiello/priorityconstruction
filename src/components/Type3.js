import { TYPE_SIZE } from '../styles/vars';
import typeComponentFactory from '../utils/typeComponentFactory';

const result = typeComponentFactory(TYPE_SIZE.t3, {
  defaultTag: 'h3',
  displayName: 'Type3',
});

export const Type3 = result.Type;

export default result.StyledType;
