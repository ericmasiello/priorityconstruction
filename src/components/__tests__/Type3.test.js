import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import Type3, { Type3 as Raw } from '../Type3';

it('should render', () => {
  const component = renderer.create(<Type3 />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render children', () => {
  const wrapper = shallow(<Raw>Test</Raw>);

  expect(wrapper.text()).toBe('Test');
});

it('should render additional props', () => {
  const wrapper = shallow(<Raw data-test="my-attribute" />);

  expect(wrapper.find('[data-test="my-attribute"]')).toHaveLength(1);
});

it('should display additional classes', () => {
  const wrapper = shallow(<Raw className="my-class" />);

  expect(wrapper.hasClass('my-class')).toBe(true);
});

it('should render as a <h3> by default', () => {
  const wrapper = shallow(<Raw />);

  expect(wrapper.type()).toBe('h3');
});

it('should render with a custom tag', () => {
  const wrapper = shallow(<Raw tag="span" />);

  expect(wrapper.type()).toBe('span');
});
