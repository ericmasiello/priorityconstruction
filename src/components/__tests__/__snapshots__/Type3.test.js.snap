// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
.c0 {
  font-size: 10vw;
  line-height: 1.3;
}

@media (max-width:10rem) {
  .c0 {
    font-size: 1rem;
  }
}

@media (min-width:17.76889rem) {
  .c0 {
    font-size: 1.776889rem;
  }
}

<h3
  className="c0"
/>
`;
